🎨 Added DID as a valid Bluesky username pattern

troyciesco opened the pull request:
closes https://linear.app/ghost/issue/PROD-1870/allow-perma-links-in-validation-for-bluesky
reported here https://forum.ghost.org/t/bluesky-entry-on-profile-social-links-blocks-perma-links/57712

- In addition to regular handles, Bluesky also has the concept of Decentralized Identifiers (DID) as noted in the [Bluesky docs](https://docs.bsky.app/docs/advanced-guides/resolving-identities).
- This update makes it so staff users can set their Bluesky username on Ghost to a DID
- also includes tidying specifically impacted by these changes

troyciesco committed:
🎨 Added DID as valid Bluesky username pattern

troyciesco committed:
adds reusable bsky username checker

troyciesco committed:
reusable errors

kevinansfield reviewed (approved):
Looks good 🙌  CodeRabbit does raise a good point about the DID usernames needing to be lowercase so updating the regex sounds like a plan. Maybe we could also do a lowercase conversion if it otherwise matches?

troyciesco committed:
add formatting step

troyciesco committed:
expand tests

troyciesco commented:
> Looks good 🙌 CodeRabbit does raise a good point about the DID usernames needing to be lowercase so updating the regex sounds like a plan. Maybe we could also do a lowercase conversion if it otherwise matches?

@kevinansfield oh yeah good point by you and coderabbit - i'd started from the suggested regex in the [atproto docs](https://atproto.com/specs/did#at-protocol-did-identifier-syntax) and modified to match bluesky's did:plc: version, but can take it even farther.

updated that, and added [formatUsername](https://github.com/TryGhost/Ghost/pull/24229/files#diff-b188ac7b6b37dd763818aeabc0353d2b293ebadfecbc42221b120492f1edab72R22) to take care of that where it's needed 👍
